### DISCUSSION TITLE
cRIO/LinuxRT support for gRPC

### DISCUSSION THREAD

**Discussion by tlhanken** [ni/grpc-device#1141] [Q&A]
Hey Team!

I have a gRPC tool built with the toolkit that works on LabVIEW on windows, but struggling to get it to run on cRIO.  There seems to be dcoumentation that people have gotten this to work on cRIOs before, but I see little documentation on the process.  How does one go about getting gRPC LabVIEW code running/setup on a NI LinuxRT?  Thanks!

**Comment by corneydavid** [comment]
We are working with a user get their product tied into LabVIEW code. Under the hood it uses gRPC. We currently have things in a working state on Windows, however, are struggling when running it on a LinuxRT target – when trying to deploy on the RT target we are getting errors for missing dependencies. Text below is from deploy log.

Deploying System [Exec.vi](https://exec.vi/) (12.33 K)
Deploying Sift Client.lvlib:Sift Client.lvclass:Get Stop Notifier [Status.vi](https://status.vi/) (6.04 K)
Deploying Sift Client.lvlib:Sift Client.lvclass:[Inititalize.vi](https://inititalize.vi/) (29.84 K)
Deploying Sift Client.lvlib:Sift Client.lvclass:[Destroy.vi](https://destroy.vi/) (11.60 K)
Deploying Sift Proxy Smoketest.viSift Proxy [Smoketest.vi](https://smoketest.vi/) loaded with errors on the target and was closed.
LabVIEW: (Hex 0x436) Failed to load shared library lvwebclient_*.*:http_Close:C  . Ensure that the library is present on the RT target.
LabVIEW: (Hex 0x436) Failed to load shared library lvwebclient_*.*:http_GetResult:C  . Ensure that the library is present on the RT target.
LabVIEW: (Hex 0x436) Failed to load shared library lvwebclient_*.*:http_Get:C  . Ensure that the library is present on the RT target.
LabVIEW: (Hex 0x436) Failed to load shared library lvwebclient_*.*:http_GetResult:C  . Ensure that the library is present on the RT target.
LabVIEW: (Hex 0x436) Failed to load shared library lvwebclient_*.*:http_Delete:C  . Ensure that the library is present on the RT target.
LabVIEW: (Hex 0x436) Failed to load shared library lvwebclient_*.*:http_Head:C  . Ensure that the library is present on the RT target.
LabVIEW: (Hex 0x436) Failed to load shared library lvwebclient_*.*:http_GetResult:C  . Ensure that the library is present on the RT target.
LabVIEW: (Hex 0x436) Failed to load shared library lvwebclient_*.*:http_AddHeader:C  . Ensure that the library is present on the RT target.
LabVIEW: (Hex 0x436) Failed to load shared library lvwebclient_*.*:http_RemoveHeader:C  . Ensure that the library is present on the RT target.
LabVIEW: (Hex 0x436) Failed to load shared library lvwebclient_*.*:http_Post:C  . Ensure that the library is present on the RT target.
LabVIEW: (Hex 0x436) Failed to load shared library lvwebclient_*.*:http_GetResult:C  . Ensure that the library is present on the RT target.
LabVIEW: (Hex 0x436) Failed to load shared library lvwebclient_*.*:http_PostFile:C  . Ensure that the library is present on the RT target.
LabVIEW: (Hex 0x436) Failed to load shared library lvwebclient_*.*:http_GetResult:C  . Ensure that the library is present on the RT target.
LabVIEW: (Hex 0x436) Failed to load shared library lvwebclient_*.*:http_PostMultipart:C  . Ensure that the library is present on the RT target.
LabVIEW: (Hex 0x436) Failed to load shared library lvwebclient_*.*:http_GetResult:C  . Ensure that the library is present on the RT target.
LabVIEW: (Hex 0x436) Failed to load shared library lvwebclient_*.*:http_GetResult:C  . Ensure that the library is present on the RT target.
LabVIEW: (Hex 0x436) Failed to load shared library lvwebclient_*.*:http_GenericRequest:C  . Ensure that the library is present on the RT target.
LabVIEW: (Hex 0x436) Failed to load shared library lvwebclient_*.*:http_Put:C  . Ensure that the library is present on the RT target.
LabVIEW: (Hex 0x436) Failed to load shared library lvwebclient_*.*:http_GetResult:C  . Ensure that the library is present on the RT target.
LabVIEW: (Hex 0x436) Failed to load shared library lvwebclient_*.*:http_PutFile:C  . Ensure that the library is present on the RT target.
LabVIEW: (Hex 0x436) Failed to load shared library lvwebclient_*.*:http_GetResult:C  . Ensure that the library is present on the RT target.
LabVIEW: (Hex 0x436) Failed to load shared library lvwebclient_*.*:http_Open:C  . Ensure that the library is present on the RT target.
Deploying RT CompactRIO Target Container
Deployment completed with errors

Is there certain files or installs needed to get LabVIEW gRPC support on a LinuxRT device? Any guidance would be appreciated, thanks!

**Reply by jasonmreding** [reply]
What LabVIEW toolkit is being used on Windows? My guess is it doesn't support RT? Typically, the answer for integrating gRPC with LV is to use [grpc-labview](https://github.com/ni/grpc-labview). Is that something you have explored?

From the little I can tell from the post above, there seems to be a "lvwebclient" library being used that is missing. I'm not sure what that library corresponds to, but it doesn't appear to be related to grpc-device itself. It sounds like it is some other library being used within the LV application code that might not be ported to RT? From the names of things, it sounds like it might be a web service/server API of some sort. Are you using standard web APIs in addition to gRPC for this application? If so, are those NI libraries/APIs or some third party APIs?

**Reply by jasonmreding** [reply]
It appears this might be at the heart of the issue:
https://forums.vipm.io/topic/8309-deploy-to-ni-linux-rt/
